File: libs/grpc/clients/emotes_cacher.go
package clients

import (
	"log"

	"github.com/twirapp/twir/libs/grpc/constants"
	"github.com/twirapp/twir/libs/grpc/emotes_cacher"

	"google.golang.org/grpc"
)

func NewEmotesCacher(env string) emotes_cacher.EmotesCacherClient {
	serverAddress := createClientAddr(env, "emotes-cacher", constants.EMOTES_CACHER_SERVER_PORT)

	conn, err := grpc.Dial(
		serverAddress,
		defaultClientsOptions...,
	)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	c := emotes_cacher.NewEmotesCacherClient(conn)
	return c
}

File: libs/grpc/clients/emotes_cacher.ts
import { ChannelCredentials, createChannel, createClient } from 'nice-grpc';

import { CLIENT_OPTIONS, createClientAddr, waitReady } from './helper.js';
import { PORTS } from '../constants/constants.js';
import {
	EmotesCacherClient,
	EmotesCacherDefinition,
} from '../emotes_cacher/emotes_cacher.client.js';

export const createEmotesCacher = async (env: string): Promise<EmotesCacherClient> => {
	const channel = createChannel(
		createClientAddr(env, 'emotes-cacher', PORTS.EMOTES_CACHER_SERVER_PORT),
		ChannelCredentials.createInsecure(),
		CLIENT_OPTIONS,
	);

	await waitReady(channel);

	const client = createClient(EmotesCacherDefinition, channel);

	return client as any;
};
